File: java/compcoding/src/main/java/com/kartik/compcoding/TreeNode.java
package com.kartik.compcoding;


import com.sun.source.tree.Tree;

public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x, TreeNode left, TreeNode right) { val = x; this.left = left; this.right = right; }
}

File: javascript/comp-coding/lib/TreeNode.js
export default function TreeNode(val = 0, right = null, left = null) {
    const inorderTraversal = (root, traversal = []) => {
        if (root) {
            if (root.left) {
                inorderTraversal(root.left, traversal);
            }
            traversal.push(root.val);
            if (root.right) {
                inorderTraversal(root.right, traversal);
            }
        }
      
        return traversal;
    }
    Object.assign(this, { val, right, left, inorderTraversal });
    return this
}